dp-game-webapp/app/game/slider.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useState } from 'react';
+
+export function CampaignSizeSlider() {
+  const [sliderExpValue, setSliderExpValue] = useState(6);
+
+  const handleSliderChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const val = parseInt(e.target.value, 10);
+    setSliderExpValue(val);
+  };
+  const displayValue = Math.pow(10, sliderExpValue).toLocaleString();
+
+  return (
+    <>
+      <p>Campaign Size (impressions): {displayValue}</p>
+      <input
+        type="range"
+        min="4"
+        max="8"
+        value={sliderExpValue}
+        onChange={handleSliderChange}
+        className="w-full"
+      />
+    </>
+  );
+}
+
+export function ConversionRateSlider() {
+  const [sliderValue, setSliderValue] = useState(2);
+
+  const handleSliderChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const val = parseInt(e.target.value, 10);
+    setSliderValue(val);
+  };
+
+  const displayValue = (5 /1000 * sliderValue).toFixed(3);
```

Comment:
The conversion rate range looks too small from 0-1. We would need to think an appropriate range for this

Reply:
I think half a percent is probably the right lower bound. 100% is probably too high, but I'm not sure where else to cut it off

Reply:
oh I thought these were absolute conversion values. Understood now. Lets change it to conversions per thousand impressions (similar to CPM) so easier to read?